### PR TITLE
Update XNNPACK to 1ed874e65

### PR DESCRIPTION
**Pull request by mcr229**
As Title

### CODE REVIEW

File: .ci/scripts/test_llama.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@ BUILD_TOOL=$2 # buck2 or cmake
 DTYPE=$3 # fp16, bf16, or fp32
 MODE=${4:-"xnnpack+custom"} # portable or xnnpack+custom or xnnpack+custom+qe
 UPLOAD_DIR=${5:-}
+CMAKE_BUILD_TYPE=${6:-Release}
```
Comment (larryliu0820) [inline .ci/scripts/test_llama.sh:17]: Can you get this from ENV

File: .ci/scripts/test_llama.sh [hunk 2]
```diff
@@ -14,6 +14,7 @@ BUILD_TOOL=$2 # buck2 or cmake
 DTYPE=$3 # fp16, bf16, or fp32
 MODE=${4:-"xnnpack+custom"} # portable or xnnpack+custom or xnnpack+custom+qe
 UPLOAD_DIR=${5:-}
+CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-Release}
```
Comment (digantdesai) [inline .ci/scripts/test_llama.sh:17]: why do we need make Release the new default?

File: .github/workflows/trunk.yml [hunk 3]
```diff
@@ -290,7 +290,7 @@ jobs:
   #       ${CONDA_RUN} python -m unittest examples.models.llava.test.test_llava
 
   #       # run e2e (export, tokenizer and runner)
-  #       PYTHON_EXECUTABLE=python ${CONDA_RUN} bash .ci/scripts/test_llava.sh Release
+  #       PYTHON_EXECUTABLE=python ${CONDA_RUN} bash .ci/scripts/test_llava.sh
```
Comment (digantdesai) [inline .github/workflows/trunk.yml:293]: oh it was Release, keep debug default in the script and pass this as is?

File: backends/xnnpack/third-party/xnnpack.buck.bzl [hunk 4]
```diff
@@ -67,7 +67,7 @@ def define_xnnpack():
     # @lint-ignore BUCKLINT: native and fb_native are explicitly forbidden in fbcode.
     native.cxx_library(
         name = "subgraph",
-        srcs = SUBGRAPH_SRCS,
+        srcs = SUBGRAPH_SRCS + ["XNNPACK/src/datatype.c"],
```
Comment (digantdesai) [inline backends/xnnpack/third-party/xnnpack.buck.bzl:70]: fix this upstream?

**Review by digantdesai — APPROVED** [review]

**Review by digantdesai — APPROVED** [review]

**Review by huydhn — APPROVED** [review]

**Comment by swolchok** [comment]
looks like this is good to go? failures look like 1 flake, 1 forked PR problem, 1 unclear

**Comment by mcr229** [comment]
yea sorry i've been trying to land this by rebasing and importing, but from internal it keeps failing because of some phantom merge conflict.

**Comment by huydhn** [comment]
@mcr229 Let me try to land this, the diff signals complaining about PR cannot be merged looks weird

**Pull request merged**